client/src/components/DetailedBookCard.js

```diff
@@ -1,34 +1,73 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { detailedBookCard } from '../assets/css/makeStyles';
 import { Paper, Box, Grid, Chip, Typography } from '@material-ui/core';
+import ReserveModal from './ReserveModal';
+import LendingModal from './LendingModal';
+import ReturnModal from './ReturnModal';
+import { useAuth } from '../lib/auth';
+import bookImage from '../assets/img/book.png';
 
 const DetailedBookCard = (props) => {
+  const { auth } = useAuth();
   const classes = detailedBookCard();
-  const { book } = props;
+  const [book, setBook] = useState(props.book);
+
+  const handleStatus = () => {
+    if (book.idUserReserve && book.idUserReserve === auth?.user?._id) {
+      if (book.status === 'Reservado')
+        return <LendingModal bookId={book._id} />;
+      if (book.status === 'Emprestado')
+        return <ReturnModal bookId={book._id} />;
+    } else if (book.status === 'Reservado') {
+      return <div>avise quando disponível</div>;
```

Comment:
Essa string é exibida ao usuário? Talvez formatar ela melhor
```suggestion
      return <div>Avise quando disponível.</div>;
```